core/src/test/java/hudson/FunctionsTest.java

```diff
@@ -639,4 +641,39 @@ synchronized Stack suppressed(Throwable... suppressed) {
             return this;
         }
     }
+
+    @Test
+    public void tryGetIcon_shouldReturnNullForNull() throws Exception {
+        assertThat(Functions.tryGetIcon(null), is(nullValue()));
+    }
+
+    @Test
+    public void tryGetIcon_shouldReturnNullForSymbol() throws Exception {
+        assertThat(Functions.tryGetIcon("symbol-search"), is(nullValue()));
+    }
+
+    @Test
+    public void tryGetIcon_shouldReturnMetadataForExactSpec() throws Exception {
+        assertThat(Functions.tryGetIcon("icon-help icon-sm"), is(not(nullValue())));
+    }
+
+    @Test
+    public void tryGetIcon_shouldReturnMetadataForExtraSpec() throws Exception {
+        assertThat(Functions.tryGetIcon("icon-help icon-sm extra-class"), is(not(nullValue())));
+    }
```

Comment:
This will fail without the change of `tryGetIcon`:

```
[ERROR] tryGetIcon_shouldReturnMetadataForExtraSpec  Time elapsed: 0.018 s  <<< FAILURE!
java.lang.AssertionError:

Expected: is not null
     but: was null
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at hudson.FunctionsTest.tryGetIcon_shouldReturnMetadataForExtraSpec(FunctionsTest.java:662)
```